it/src/main/java/org/corfudb/universe/node/server/docker/DockerSupportServer.java

```diff
@@ -0,0 +1,175 @@
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.docker.client.messages.PortBinding;
+import groovy.util.logging.Log4j;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.commons.lang.StringUtils;
+import org.corfudb.universe.group.cluster.SupportClusterParams;
+import org.corfudb.universe.node.Node;
+import org.corfudb.universe.node.NodeException;
+import org.corfudb.universe.node.server.SupportServer;
+import org.corfudb.universe.universe.UniverseParams;
+import org.corfudb.universe.util.DockerManager;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+@Log4j
+public class DockerSupportServer<N extends Node.NodeParams> implements SupportServer {
+    private static final String ALL_NETWORK_INTERFACES = "0.0.0.0";
+    private static final Map<NodeType, String> imageName = ImmutableMap.<NodeType, String>builder()
+            .put(NodeType.METRICS_SERVER, "prom/prometheus")
+            .put(NodeType.SHELL_NODE, "ubuntu")
+            .build();
+
+    private static final Map<NodeType, String> cmd = ImmutableMap.<NodeType, String>builder()
+            .put(NodeType.SHELL_NODE, "bash")
+            .build();
+
+    @Getter
+    protected final N params;
+
+    @NonNull
+    @Getter
+    protected final UniverseParams universeParams;
+
+    @NonNull
+    private final DockerClient docker;
+
+    @NonNull
+    private final DockerManager dockerManager;
+
+    @NonNull
+    private final SupportClusterParams clusterParams;
+
+    @NonNull
+    private final AtomicReference<String> ipAddress = new AtomicReference<>();
+
+    @Builder
+    public DockerSupportServer(DockerClient docker, DockerManager dockerManager,
+                               N params,
+                               SupportClusterParams clusterParams, UniverseParams universeParams) {
+        this.docker = docker;
+        this.dockerManager = dockerManager;
+        this.params = params;
+        this.clusterParams = clusterParams;
+        this.universeParams = universeParams;
+    }
+
+    @Override
+    public SupportServer deploy() {
+        deployContainer();
+
+        return this;
+    }
+
+    private ContainerConfig buildContainerConfig() {
+        List<String> ports = params.getPorts().stream()
+                .map(Objects::toString).collect(Collectors.toList());
+        Map<String, List<PortBinding>> portBindings = new HashMap<>();
+        for (String port : ports) {
+            List<PortBinding> hostPorts = new ArrayList<>();
+            hostPorts.add(PortBinding.of(ALL_NETWORK_INTERFACES, port));
+            portBindings.put(port, hostPorts);
+        }
+
+        HostConfig.Bind configurationFile =  HostConfig.Bind.builder()
+                .from("/home/nsx/prometheus.yml")
```

Comment:
don't use `nsx` dir name

Reply:
I'll remove that part for now and add is as a constructor parameter.